src/main/java/org/sagebionetworks/web/client/widget/entity/MarkdownEditorWidgetViewImpl.java

```diff
@@ -274,48 +267,25 @@ public void onClick(ClickEvent event) {
 				formattingGuideModal.show();
 			}
 		});
-		
-		markdownTextArea.addClickHandler(new ClickHandler() {
-			@Override
-			public void onClick(ClickEvent event) {
-				presenter.markdownEditorClicked();
-			}
-		});
-		
-		markdownTextArea.addKeyUpHandler(new KeyUpHandler() {
-			@Override
-			public void onKeyUp(KeyUpEvent event) {
-				presenter.markdownEditorClicked();
-			}
-		});
-		markdownTextArea.addKeyUpHandler(new KeyUpHandler() {
-			@Override
-			public void onKeyUp(KeyUpEvent event) {
-				resizeMarkdownTextArea(0);
-			}
-		});
 	}
 	
-	private ClickHandler getClickHandler(final MarkdownEditorAction action) {
-		return new ClickHandler() {
-			@Override
-			public void onClick(ClickEvent event) {
-				presenter.handleCommand(action);		
-			}
-		};
+	public void setDeleteClickHandler(ClickHandler handler) {
+		deleteButton.addClickHandler(handler);
+	}
+	
+	public void addTextAreaKeyUpHandler(KeyUpHandler handler) {
+		markdownTextArea.addKeyUpHandler(handler);
+	}
+	
+	public void addTextAreaClickHandler(ClickHandler handler) {
+		markdownTextArea.addClickHandler(handler);
 	}
 	
-	private ClickHandler getDeleteClickHandler() {
+	private ClickHandler getClickHandler(final MarkdownEditorAction action) {

```

Comment:
getClickHandler only includes: 

	private ClickHandler getClickHandler(final MarkdownEditorAction action) {
		return new ClickHandler() {
			@Override
			public void onClick(ClickEvent event) {
				presenter.handleCommand(action);		
			}
		};
	}

And I moved the construction of the delete button click handler to the presenter.